### ISSUE TITLE
Lint/UselessConstantScoping: False positives when using `private` with arguments

### ISSUE THREAD

**Issue by colszowka** [rubocop/rubocop#13856]
Hi there,

I believe the new `Lint/UselessConstantScoping` cop produces false positives when using `private` with arguments, i.e. in a scenario like this:

```ruby
class Foo
  private attr_accessor :foo

  MY_CONST = "foo"

  def bar
    "this is not private..."
  end
end
```

--------

## Expected behavior

In the example above, `private` is used with arguments and hence does not affect the code below it. So the cop should not complain about `MY_CONST` since it's not actually offending the cop.

## Actual behavior

```
example.rb:4:3: W: Lint/UselessConstantScoping: Useless private access modifier for constant scope.
  MY_CONST = "foo"
  ^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Take my code from above and run Rubocop 1.72.1 against it.

## RuboCop version

```
1.72.1 (using Parser 3.3.7.1, rubocop-ast 1.38.0, analyzing as Ruby 2.7, running on ruby 3.4.1) [x86_64-linux]
```

Thanks!
